packages/ui-text-input/src/TextInput/props.ts

```diff
@@ -38,153 +38,186 @@ import type {
-  type?: 'text' | 'email' | 'url' | 'tel' | 'search' | 'password'
-  id?: string
-  value?: any // TODO: controllable(PropTypes.string)
-  defaultValue?: string
-  interaction?: InteractionType
-  messages?: FormMessage[]
-  size?: 'small' | 'medium' | 'large'
-  textAlign?: 'start' | 'center'
-  width?: string
-  htmlSize?: string | number
-  display?: 'inline-block' | 'block'
-  shouldNotWrap?: boolean
-  placeholder?: string
-  isRequired?: boolean
-  elementRef?: (element: Element | null) => void
-  inputRef?: (...args: any[]) => any
-  inputContainerRef?: (...args: any[]) => any
-  renderBeforeInput?: React.ReactNode | ((...args: any[]) => any)
-  renderAfterInput?: React.ReactNode | ((...args: any[]) => any)
-  onChange?: (...args: any[]) => any
-  onBlur?: (...args: any[]) => any
-  onFocus?: (...args: any[]) => any
-}
-
-type PropKeys = keyof TextInputOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type TextInputProps = TextInputOwnProps &
-  WithStyleProps<TextInputTheme, TextInputStyle> &
-  OtherHTMLAttributes<TextInputOwnProps, InputHTMLAttributes<TextInputOwnProps>>
-
-type TextInputStyle = ComponentStyle<
-  | 'textInput'
-  | 'facade'
-  | 'layout'
-  | 'beforeElement'
-  | 'innerWrapper'
-  | 'inputLayout'
-  | 'afterElement'
->
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * The form field label.
    */
-  renderLabel: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+  renderLabel?: React.ReactNode | (() => React.ReactNode)
+
   /**
    * Determines the underlying native HTML `<input>` element's `type`.
+   *
    * For more see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url
    */
-  type: PropTypes.oneOf(['text', 'email', 'url', 'tel', 'search', 'password']),
+  type?: 'text' | 'email' | 'url' | 'tel' | 'search' | 'password'
+
   /**
    * The id of the text input. One is generated if not supplied.
    */
-  id: PropTypes.string,
+  id?: string
+
   /**
    * the selected value (must be accompanied by an `onChange` prop)
    */
-  value: controllable(PropTypes.string),
+  value?: string // TODO: controllable(PropTypes.string)
+
   /**
    * value to set on initial render
    */
-  defaultValue: PropTypes.string,
+  defaultValue?: string
+
   /**
    * Specifies if interaction with the input is enabled, disabled, or readonly.
    * When "disabled", the input changes visibly to indicate that it cannot
    * receive user interactions. When "readonly" the input still cannot receive
    * user interactions but it keeps the same styles as if it were enabled.
    */
-  interaction: PropTypes.oneOf(['enabled', 'disabled', 'readonly']),
+  interaction?: InteractionType
+
   /**
-   * object with shape: `{
-   * text: PropTypes.node,
-   * type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
-   *   }`
+   * Array of objects with shape: `{
+   *   text: PropTypes.node,
+   *   type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
```

Comment:
I think this description could be more readable, something like:
Array of objects with shape: ```{
text: React.ReactNode,
type: One of  'error', 'hint', 'success', 'screenreader-only'
```